test/update_test.go

```diff
@@ -34,70 +34,63 @@ func TestUpdate(t *testing.T) {
-				If(clientv3.Compare(clientv3.ModRevision("updateExistingKey"), "=", lastModRev)).
-				Then(clientv3.OpPut("updateExistingKey", "testValue2")).
-				Else(clientv3.OpGet("updateExistingKey", clientv3.WithRange(""))).
-				Commit()
-
-			g.Expect(err).To(BeNil())
-			g.Expect(resp.Succeeded).To(BeTrue())
-		}
-
-		{
-			resp, err := client.Get(ctx, "updateExistingKey", clientv3.WithRange(""))
-			g.Expect(err).To(BeNil())
-			g.Expect(resp.Kvs).To(HaveLen(1))
-			g.Expect(resp.Kvs[0].Key).To(Equal([]byte("updateExistingKey")))
-			g.Expect(resp.Kvs[0].Value).To(Equal([]byte("testValue2")))
-			g.Expect(resp.Kvs[0].ModRevision).To(BeNumerically(">", resp.Kvs[0].CreateRevision))
-		}
+		resp, err := client.Get(ctx, "updateExistingKey", clientv3.WithRange(""))
+		g.Expect(err).To(BeNil())
+		g.Expect(resp.Kvs).To(HaveLen(1))
+		g.Expect(resp.Kvs[0].Key).To(Equal([]byte("updateExistingKey")))
+		g.Expect(resp.Kvs[0].Value).To(Equal([]byte("testValue2")))
+		g.Expect(resp.Kvs[0].ModRevision).To(BeNumerically(">", resp.Kvs[0].CreateRevision))
 	})
 
 	// Trying to update an old revision(in compare) should fail
 	t.Run("UpdateOldRevisionFails", func(t *testing.T) {
 		g := NewWithT(t)
 
 		lastModRev := createKey(ctx, g, client, "updateOldRevKey", "testValue1")
+		updateRev(ctx, g, client, "updateOldRevKey", lastModRev, "testValue2")
+
+		resp, err := client.Txn(ctx).
+			If(clientv3.Compare(clientv3.ModRevision("updateOldRevKey"), "=", lastModRev)).
+			Then(clientv3.OpPut("updateOldRevKey", "testValue2")).
+			Else(clientv3.OpGet("updateOldRevKey", clientv3.WithRange(""))).
+			Commit()
 
-		{
-			resp, err := client.Txn(ctx).
-				If(clientv3.Compare(clientv3.ModRevision("updateOldRevKey"), "=", lastModRev)).
-				Then(clientv3.OpPut("updateOldRevKey", "testValue2")).
-				Else(clientv3.OpGet("updateOldRevKey", clientv3.WithRange(""))).
-				Commit()
-
-			g.Expect(err).To(BeNil())
-			g.Expect(resp.Succeeded).To(BeTrue())
-		}
-
-		{
-			resp, err := client.Txn(ctx).
-				If(clientv3.Compare(clientv3.ModRevision("updateOldRevKey"), "=", lastModRev)).
-				Then(clientv3.OpPut("updateOldRevKey", "testValue2")).
-				Else(clientv3.OpGet("updateOldRevKey", clientv3.WithRange(""))).
-				Commit()
-
-			g.Expect(err).To(BeNil())
-			g.Expect(resp.Succeeded).To(BeFalse())
-			g.Expect(resp.Responses).To(HaveLen(1))
-			g.Expect(resp.Responses[0].GetResponseRange()).ToNot(BeNil())
-		}
+		g.Expect(err).To(BeNil())
+		g.Expect(resp.Succeeded).To(BeFalse())
+		g.Expect(resp.Responses).To(HaveLen(1))
+		g.Expect(resp.Responses[0].GetResponseRange()).ToNot(BeNil())
 	})
 }
 
-func updateEntry(ctx context.Context, g Gomega, client *clientv3.Client, key string, value string) {
+// BenchmarkUpdate is a benchmark for the Update operation.
+func BenchmarkUpdate(b *testing.B) {
+	b.StopTimer()
+	g := NewWithT(b)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	client, _ := newKine(ctx, b)
+
+	b.StartTimer()
```

Comment:
The timer is not explicitly stopped, is it?

Reply:
It is on line 69, isn't it? Am I misunderstanding this comment?